src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -101,6 +101,27 @@ public CertificateInfo getCertificateInfo(String hash) throws CertificateNotFoun
         return certificateInfo;
     }
 
+    /**
+     * Find an existing cert by it's hash and import
+     * @param hash cert hash of an existing cert
+     * @return CertificateType
+     * @throws CertificateNotFoundException
+     * @throws InvalidCertificateException other general import failure
+     * @throws GlobalConfService.GlobalConfOutdatedException
+     * @throws KeyNotFoundException
+     * @throws CertificateAlreadyExistsException
+     * @throws WrongCertificateUsageException
+     * @throws ClientNotFoundException
+     * @throws CsrNotFoundException
+     */
+    public CertificateType importExistingCertificate(String hash) throws CertificateNotFoundException,
+            InvalidCertificateException, GlobalConfService.GlobalConfOutdatedException, KeyNotFoundException,
```

Comment:
This has now been addressed about around [here](https://github.com/nordic-institute/X-Road/blob/XRDDEV-707-import-existing-cert/src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java#L157).
So now instead of showing a string `auth, not supported` on frontend we throw an exception if an auth cert is being imported from a token.